### DISCUSSION TITLE
Prefetch Cache

### DISCUSSION THREAD

**Discussion by barbinbrad** [remix-run/remix#4317] [Q&A] [answered]
How does`prefetch=intent` work? I noticed that it's fired off every time I hover. Are there any mechanisms for caching the prefetch so it doesn't fire off twice within an X second window?

**Comment by sergiodxa** [comment, marked as answer]
You should cache the prefetched loader data so the next time is cached or if it’s expired it’s prefetched again on hover

**Reply by barbinbrad** [reply]
Oh man, I was just coming back to say I've read your article. And it was very helpful. Thanks for the quick response.

Regarding cache-control headers, it seems easy enough for unauthenticated requests. But what about when multiple people could be requesting the same endpoint and getting different results? Is the idea then to add some type of requester id param to the URL?

**Reply by kiliman** [reply]
```ts
export let loader: LoaderFunction = async ({ request }) => {
  let data = await getData(request);
  return json(data, {
    headers: { "Cache-Control": "private, max-age=10" },
  });
};
```
The caching is done locally in the browser for that specific user. Are you thinking of CDN caches? It would be something like `"Cache-Control": "public, s-max-age=10"`. You should not include those headers on private data.

**Reply by sergiodxa** [reply]
☝️ this, use a private cache for a few seconds only on the user browser, then you can cache authenticated requests without issues, the `private` ensures it's only cached by the browser, you may also want to add `s-maxage=0` to be extra safe that CDNs will not cache it

**Reply by barbinbrad** [reply]
Ahh, I see. Thanks, gentlemen!
